addons/account/models/account_move.py

```diff
@@ -854,7 +854,11 @@ def auto_reconcile_lines(self):
         elif self._context.get('skip_full_reconcile_check') == 'amount_currency_only':
             currency = self._context.get('manual_full_reconcile_currency')
 
-        self.env['account.partial.reconcile'].create({
+        skip_full_reconcile_check = self._context.get('skip_full_reconcile_check')
+        if not sm_debit_move.currency_id and not sm_credit_move.currency_id:
+            skip_full_reconcile_check = True
+
+        self.env['account.partial.reconcile'].with_context(skip_full_reconcile_check=skip_full_reconcile_check).create({
```

Comment:
That would make more sense. Interestingly enough however I just tried it and it doesn't help performance at all for some reason which is weird. Somehow this must be triggering more expensive computations compared to my original code... I will have to investigate further.